confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/macros/AbstractPageMacroConverter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.confluence.filter.internal.macros;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Add the page parameter to converted macro.
+ *
+ * @version $Id$
+ * @since 9.26.0
+ */
+public abstract class AbstractPageMacroConverter extends AbstractMacroConverter
+{
+    @Override
+    protected Map<String, String> toXWikiParameters(String confluenceId, Map<String, String> confluenceParameters,
+        String content)
+    {
+        Map<String, String> parameters = new LinkedHashMap<>(1);
+        parameters.put("page", confluenceParameters.get(""));
```

Comment:
"page" parameter is based on [page reference](https://extensions.xwiki.org/xwiki/bin/view/Extension/Model%20Module#HPageReferences) syntax (so with a "/" separator) but in the syntax parser you seem to be using "." as separator (so document reference syntax) so this won't work if there is space information.

Also, you need to go through `ConfluenceInputFilterStream#convert(String entityReference, EntityType entityType)` to take into account things like the entity name validator.

Reply:
Honestly, I'm not sure what's best. A pity the value is a String, so we have to choose something.

I guess the safest default behavior for the syntax parser is what you did, which mean changing the converter to assume it's a document reference and not a page reference.

Reply:
> Shouldn't this be done for regular links then?

It's the case, as you can see from the `links` test I mentioned.

> Do you know how I could access this method to this part of the code? It's in a different package, confluence-xml, that depends on confluence-xhtml so depending on confluence-xml would introduce a circular dependency. I also suspect confluence-xhtml could be used outside the context of confluence-xml.

Not sure why you mention confluence-xhtml, macro converters are a confluence-xml thing. But we do need some way to make it easier for macro converter to convert reference, as it's not really passed to MacroConverter#toXWiki.

> Have we just found a bug here? Should confluence-xml pass mappings / validators to confluence-xhtml?

Not sure what you mean, reference conversion is not the syntax parser's job.

Reply:
OK, I think you assumed I was suggesting you to convert the reference in the syntax parser in my first comment, but I was talking about the macro converter.